rust/arrow/src/array/array_primitive.rs

```diff
@@ -112,6 +112,22 @@ impl<T: ArrowPrimitiveType> PrimitiveArray<T> {
         );
         PrimitiveArray::from(data)
     }
+
+    /// Creates a PrimitiveArray based on a constant value with `count` elements
+    pub fn from_value(value: T::Native, count: usize) -> Self {
```

Comment:
What do you think of making it `from_trusted_len_values_iter` or something like that, and pass `(0..count).map(|_| value)` to it? This way we do not restrict the function to the specific use-case where the value is constant, and allow more complex behavior to the user.

Reply:
I can see the benefits, the downside of this would be exposing an extra `unsafe` method to the arrow crate.

Reply:
`unsafe` is not inherently bad, it is just used to flag developers that an API requires assumptions beyond what the compiler can prove.

This crates' whole design embraces unsafe public APIs not marked as `unsafe`, which is bad, so this seems negligible to me.

My understanding is that we need this `unsafe` because `TrustedLen` is still unstable.

Reply:
You are right, and I think it would be nice to use the `TrustedLen` trait once that is stable :).

Reply:
FWIW, I tried introducing `unsafe trait TrustedLen` on this crate and arrow2, so that we could declare `fn from_trusted_len<I: TrustedLen + Iterator>`, but I got hit by another unstability.

Essentially `0..10` is a `Range<A: Step>` and `Step` is `unstable`, which means that we can't implement our own `impl<A: Step> TrustedLen for Range<A: Step>` to enable ranged iterators on it. This is being tracked on Rust's side here: https://github.com/rust-lang/rust/issues/42168